### PR TITLE
Don't emit prometheus metrics for metrics that are disabled

### PR DESCRIPTION
**Pull request by dashpole**
Not all of the metrics can actually be ignored, but I found it helpful to sort them by the MetricKind.
Fixes https://github.com/google/cadvisor/issues/1925

### CODE REVIEW

**Comment by tallclair** [comment]
Should we make this a whitelist instead of a blacklist?

**Comment by dashpole** [comment]
We can change the flag to a whitelist from a blacklist in a separate PR if we want.   This PR should make prometheus more use-able for those who are ignoring metrics, and I would like to add this soon.

**Comment by tallclair** [comment]
I meant that maybe the `NewPrometheusCollector` function should take a whitelist. If the plan is to call this from Kubernetes, I think that makes more sense (so we don't need to update k8s for metrics it's not using). WDYT?

**Comment by dashpole** [comment]
We should probably do that for both `NewPrometheusCollector` and [manager.New](https://github.com/google/cadvisor/blob/master/manager/manager.go#L145).  Ill can change this PR to do that.

**Comment by tallclair** [comment]
test failure looks legit? Other than that, lgtm.

File: container/docker/handler.go [hunk 1]
```diff
@@ -345,14 +345,14 @@ func (self *dockerContainerHandler) ContainerReference() (info.ContainerReferenc
...
-	if !self.ignoreMetrics.Has(container.NetworkUsageMetrics) {
+	if self.includedMetrics.Has(container.NetworkUsageMetrics) {
 		return !self.networkMode.IsContainer()
 	}
 	return false
 }
 
 func (self *dockerContainerHandler) GetSpec() (info.ContainerSpec, error) {
-	hasFilesystem := !self.ignoreMetrics.Has(container.DiskUsageMetrics)
+	hasFilesystem := !self.includedMetrics.Has(container.DiskUsageMetrics)
```
Comment (dashpole) [inline container/docker/handler.go]: Test failed because I forgot to remove the `!` here.

**Comment by bboreham** [comment]
Do you need someone to port this into kubelet?

**Pull request merged**